src/utils/config.py

```
@@ -62,3 +89,63 @@ class Settings(BaseSettings):
+        self._model = self._get_default_model_for_provider(self._provider)
+        # Apply environment override via setter (validates and aligns model)
+        self.provider = settings.GENAI_PROVIDER
+
+    @property
+    def provider(self) -> str:
+        return self._provider
+
+    @provider.setter
+    def provider(self, value: str) -> None:
+        if value not in ["openai", "google"]:
+            raise ValueError(f"Invalid provider: {value}. Must be 'openai' or 'google'")
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)